### PR TITLE
Add net.zdechov.app.Tunneler

### PR DESCRIPTION
**Pull request by chronoscz**
### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
This is a remake of the famous MS-DOS Tunneler game originally written in Pascal language. [​https://tunneler.org/](https://tunneler.org/) Now it has been written from scratch in Lazarus/FPC to run natively on Linux.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: net.zdechov.app.Tunneler.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - install -m 755 Install/flatpak/${FLATPAK_ID}.desktop $FLATPAK_DEST/share/applications
+      - install -d -m 755 $FLATPAK_DEST/share/icons/hicolor/256x256/apps
+      - install -m 644 Images/256x256/Tunneler.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -d -m 755 $FLATPAK_DEST/share/Tunneler/Languages
+      - install -D -m 755 Languages/*.pot $FLATPAK_DEST/share/Tunneler/Languages
+      - install -D -m 755 Languages/*.po $FLATPAK_DEST/share/Tunneler/Languages
+      - install -d -m 755 $FLATPAK_DEST/share/Tunneler/Audio
+      - install -D -m 755 Audio/*.wav $FLATPAK_DEST/share/Tunneler/Audio
+      - install -d -m 755 $FLATPAK_DEST/share/metainfo
+      - install -D Install/flatpak/${FLATPAK_ID}.appdata.xml $FLATPAK_DEST/share/metainfo/${FLATPAK_ID}.metainfo.xml
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:47]:
```suggestion
      - install -Dm744 Install/flatpak/${FLATPAK_ID}.appdata.xml $FLATPAK_DEST/share/metainfo/${FLATPAK_ID}.metainfo.xml
```

File: net.zdechov.app.Tunneler.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    build-commands:
+      - (. /usr/lib/sdk/freepascal/enable.sh && lazbuild --build-mode=Release --ws=qt5 tunneler.lpi)
+      - install -d -m 755 $FLATPAK_DEST/share/Tunneler
+      - install -s -m 755 Tunneler $FLATPAK_DEST/share/Tunneler
+      - install -d -m 755 $FLATPAK_DEST/bin
+      - ln -s $FLATPAK_DEST/share/Tunneler/Tunneler $FLATPAK_DEST/bin/Tunneler
+      - install -d -m 755 $FLATPAK_DEST/share/applications
+      - install -m 755 Install/flatpak/${FLATPAK_ID}.desktop $FLATPAK_DEST/share/applications
+      - install -d -m 755 $FLATPAK_DEST/share/icons/hicolor/256x256/apps
+      - install -m 644 Images/256x256/Tunneler.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:40]:
```suggestion
      - install -Dm644 Images/256x256/Tunneler.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
```

File: net.zdechov.app.Tunneler.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        revision: r99
+    buildsystem: simple
+    build-commands:
+      - (. /usr/lib/sdk/freepascal/enable.sh && lazbuild --build-mode=Release --ws=qt5 tunneler.lpi)
+      - install -d -m 755 $FLATPAK_DEST/share/Tunneler
+      - install -s -m 755 Tunneler $FLATPAK_DEST/share/Tunneler
+      - install -d -m 755 $FLATPAK_DEST/bin
+      - ln -s $FLATPAK_DEST/share/Tunneler/Tunneler $FLATPAK_DEST/bin/Tunneler
+      - install -d -m 755 $FLATPAK_DEST/share/applications
+      - install -m 755 Install/flatpak/${FLATPAK_ID}.desktop $FLATPAK_DEST/share/applications
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:38]:
```suggestion
      - install -Dm644 Install/flatpak/${FLATPAK_ID}.desktop $FLATPAK_DEST/share/applications
```

File: net.zdechov.app.Tunneler.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    sources:
+      - type: svn
+        url: https://svn.zdechov.net/Tunneler/tags/1.1.1
+        revision: r99
+    buildsystem: simple
+    build-commands:
+      - (. /usr/lib/sdk/freepascal/enable.sh && lazbuild --build-mode=Release --ws=qt5 tunneler.lpi)
+      - install -d -m 755 $FLATPAK_DEST/share/Tunneler
+      - install -s -m 755 Tunneler $FLATPAK_DEST/share/Tunneler
+      - install -d -m 755 $FLATPAK_DEST/bin
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:35]:
```suggestion
      - install -d $FLATPAK_DEST/bin
```

File: net.zdechov.app.Tunneler.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  - name: Tunneler
+    sources:
+      - type: svn
+        url: https://svn.zdechov.net/Tunneler/tags/1.1.1
+        revision: r99
+    buildsystem: simple
+    build-commands:
+      - (. /usr/lib/sdk/freepascal/enable.sh && lazbuild --build-mode=Release --ws=qt5 tunneler.lpi)
+      - install -d -m 755 $FLATPAK_DEST/share/Tunneler
+      - install -s -m 755 Tunneler $FLATPAK_DEST/share/Tunneler
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:34]:
```suggestion
      - install -Dm 755 Tunneler -t $FLATPAK_DEST/share/Tunneler
```

File: net.zdechov.app.Tunneler.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        commands:
+          - cp -r /usr/lib/sdk/freepascal/share/lazarus/lcl/interfaces/qt5/cbindings/. .
+  - name: Tunneler
+    sources:
+      - type: svn
+        url: https://svn.zdechov.net/Tunneler/tags/1.1.1
+        revision: r99
+    buildsystem: simple
+    build-commands:
+      - (. /usr/lib/sdk/freepascal/enable.sh && lazbuild --build-mode=Release --ws=qt5 tunneler.lpi)
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:32]:
```suggestion
      - . /usr/lib/sdk/freepascal/enable.sh
      - lazbuild --build-mode=Release --ws=qt5 tunneler.lpi)
```

File: net.zdechov.app.Tunneler.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      - install -s -m 755 Tunneler $FLATPAK_DEST/share/Tunneler
+      - install -d -m 755 $FLATPAK_DEST/bin
+      - ln -s $FLATPAK_DEST/share/Tunneler/Tunneler $FLATPAK_DEST/bin/Tunneler
+      - install -d -m 755 $FLATPAK_DEST/share/applications
+      - install -m 755 Install/flatpak/${FLATPAK_ID}.desktop $FLATPAK_DEST/share/applications
+      - install -d -m 755 $FLATPAK_DEST/share/icons/hicolor/256x256/apps
+      - install -m 644 Images/256x256/Tunneler.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -d -m 755 $FLATPAK_DEST/share/Tunneler/Languages
+      - install -D -m 755 Languages/*.pot $FLATPAK_DEST/share/Tunneler/Languages
+      - install -D -m 755 Languages/*.po $FLATPAK_DEST/share/Tunneler/Languages
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:43]:
```suggestion
      - install -Dm644 Languages/*.pot -t $FLATPAK_DEST/share/Tunneler/Languages
      - install -Dm644 Languages/*.po -t $FLATPAK_DEST/share/Tunneler/Languages
```

Note: I don't think you need the `.pot`. They are PO templates and only relevant for translators.

File: net.zdechov.app.Tunneler.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+sdk: org.kde.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.freepascal
+command: Tunneler
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --filesystem=home
```
Comment (hfiguiere) [inline net.zdechov.app.Tunneler.yml:13]: why does it need home permissions?

Comment (chronoscz) [reply net.zdechov.app.Tunneler.yml]: It stores its settings in ~/.config/Tunneler but I guess no. Removed.

File: net.zdechov.app.Tunneler.yml (hunk 7 above)
Comment (chronoscz) [reply net.zdechov.app.Tunneler.yml]: Removed.

File: net.zdechov.app.Tunneler.yml (hunk 5 above)
Comment (chronoscz) [reply net.zdechov.app.Tunneler.yml]: Updated.

File: net.zdechov.app.Tunneler.yml (hunk 4 above)
Comment (chronoscz) [reply net.zdechov.app.Tunneler.yml]: Updated.

File: net.zdechov.app.Tunneler.yml (hunk 3 above)
Comment (chronoscz) [reply net.zdechov.app.Tunneler.yml]: Updated.

File: net.zdechov.app.Tunneler.yml (hunk 2 above)
Comment (chronoscz) [reply net.zdechov.app.Tunneler.yml]: Updated.

**Comment by chronoscz** [comment]
Thanks for the comments. They should be resolved.

File: net.zdechov.app.Tunneler.yml (hunk 6 above)
Comment (hfiguiere) [reply net.zdechov.app.Tunneler.yml]: no. one per line is better.

Comment (hfiguiere) [reply net.zdechov.app.Tunneler.yml]: nah it's ok

File: net.zdechov.app.Tunneler.yml (hunk 8 above)
Comment (hfiguiere) [reply net.zdechov.app.Tunneler.yml]: you could use `--persist` to make sure it stays. Unless it's actually using XDG directories which mean it's handled.

Comment (chronoscz) [reply net.zdechov.app.Tunneler.yml]: From the testing it looks it remembers the settings. So it is ok as is.

**Comment by bbhtt** [comment]
bot, build net.zdechov.app.Tunneler

**Comment by chronoscz** [comment]
It looks like Subversion server was not accessible during last build run. Will it re-run again automatically in some time or it is possible to re-run it manually?

**Comment by hfiguiere** [comment]
you can do it.

**Comment by chronoscz** [comment]
I see the Rebuild button on the top right but after clicking on it the message appears "Cannot rebuild: You need commit permissions to https://github.com/flathub/flathub or flathub admin status". I don't see any other way how I can do it.

**Comment by hfiguiere** [comment]
image

**Comment by chronoscz** [comment]
It was queued and then it has failed. So that means that it is still queued? Then how long it will take to run again? Is there some queue visible on web? I don't see it on buildbot web site.

**Comment by bbhtt** [comment]
bot, build net.zdechov.app.Tunneler

**Pull request closed without merging**